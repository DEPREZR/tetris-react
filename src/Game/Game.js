import React, { useContext, useRef, useEffect } from 'react';
import { InputsContext } from 'InputsListener';
import PropTypes from 'prop-types';
import { drawGameBoard } from 'canvasHelpers/canvasHelpers';
import { gameBoardDataFixtures } from 'Fixtures/gameBoardData';
import { INTERVAL_BETWEEN_CALLBACKS_TOUCHED_PRESSED } from 'constants.js';

const callbackLeftInputPressed = () => console.log('Left pressed');

const Game = ({ inputsContext = useContext(InputsContext) }) => {
  const { pressedDown, pressedLeft } = inputsContext;
  const refCanvasTetris = useRef(null);
  const intervalId = useRef(undefined);

  useEffect(() => {
    const ctx = refCanvasTetris.current.getContext('2d');

    drawGameBoard({ ctx, gameBoardData: gameBoardDataFixtures[0] });
  }, [refCanvasTetris]);

  useEffect(() => {
    if (pressedLeft) {
      callbackLeftInputPressed();
      intervalId.current = setInterval(() => {
        callbackLeftInputPressed();
      }, INTERVAL_BETWEEN_CALLBACKS_TOUCHED_PRESSED);
    } else {
      clearInterval(intervalId.current);
    }
  }, [pressedLeft]);

  return (
    <React.Fragment>
      <p>{`You are ${pressedDown ? '' : 'not'} pressing down arrow`}</p>
      <p>{`You are ${pressedLeft ? '' : 'not'} pressing left arrow`}</p>
      <canvas
        ref={refCanvasTetris}
        width={300}
        height={600}
        style={{ border: '2px solid black' }}
      />
    </React.Fragment>
  );
};

Game.propTypes = {
  inputsContext: PropTypes.object
};

export default Game;
