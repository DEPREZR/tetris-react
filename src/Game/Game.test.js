import React from 'react';
import { shallow } from 'enzyme';
import Game from '.';

describe('Game', () => {
  let minProps;

  beforeEach(() => {
    minProps = {
      inputsContext: { countDown: 3, countLeft: 2 }
    };
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Game {...minProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
