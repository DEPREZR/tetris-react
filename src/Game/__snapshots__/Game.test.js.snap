// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders correctly 1`] = `
<Fragment>
  <p>
    You are not pressing down arrow
  </p>
  <p>
    You are not pressing left arrow
  </p>
  <canvas
    height={600}
    style={
      Object {
        "border": "2px solid black",
      }
    }
    width={300}
  />
</Fragment>
`;
